pkg/ring/ring_test.go

```diff
@@ -124,16 +123,129 @@ func TestAddIngesterReplacesExistingTokens(t *testing.T) {
 	require.Equal(t, newTokens, r.Ingesters[ing1Name].Tokens)
 }
 
+func TestRing_Get_ZoneAwareness(t *testing.T) {
+	// Number of tests to run.
+	const testCount = 10000
+
+	tests := map[string]struct {
+		numIngesters         int
+		numZones             int
+		replicationFactor    int
+		zoneAwarenessEnabled bool
+		expectedErr          string
+		expectedIngesters    int
+	}{
+		"should succeed if there are enough ingesters per zone on RF = 3": {
+			numIngesters:         16,
+			numZones:             3,
+			replicationFactor:    3,
+			zoneAwarenessEnabled: true,
+			expectedIngesters:    3,
+		},
+		"should fail if there are ingesters in 1 zone only on RF = 3": {
+			numIngesters:         16,
+			numZones:             1,
+			replicationFactor:    3,
+			zoneAwarenessEnabled: true,
+			expectedErr:          "at least 2 live replicas required, could only find 1",
```

Comment:
Would it be easy for error message to mention in this case that replicas must be from different zones?

Reply:
Sure, makes sense. Done.